src/main/java/wolox/training/models/Book.java

```diff
@@ -56,14 +61,15 @@ public long getId() {
     }
 
     public void setId(long id) {
-        this.id = id;
+        this.id = checkNotNull(id, ValidationError.NULL_VALUE.getMsg());
```

Comment:
Great, a comment this is one of a lot preconditions, for example for id just be numbers, or if is a email needs to have and special character like '@', or for names not have special characters if they are not needed and for the years or birhdate that the date is in a range or something like that.